ambry-replication/src/main/java/com/github/ambry/replication/BackupCheckManager.java

```diff
@@ -0,0 +1,76 @@
+package com.github.ambry.replication;
+
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.ClusterParticipant;
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.commons.ResponseHandler;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.ReplicationConfig;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.network.ConnectionPool;
+import com.github.ambry.notification.NotificationSystem;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.MessageInfo;
+import com.github.ambry.store.StoreKeyConverter;
+import com.github.ambry.store.StoreKeyConverterFactory;
+import com.github.ambry.store.StoreKeyFactory;
+import com.github.ambry.store.Transformer;
+import com.github.ambry.utils.Utils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Predicate;
+
+
+public class BackupCheckManager extends ReplicationManager {
```

Comment:
Looks like the only reason we have `BackupCheckManager` is so that we can create `BackupCheckerThread` objects. Lets try to find a cleaner way to do the same in `ReplciationManager`. Maybe create a config for ReplicaThread class instead of ReplicationManager class?

Reply:
good idea!

Reply:
No the reason for this is to have both ReplicaThread and BAckupCheckerThread running at the same time.
CloudToBlobStoreReplicationManager would create ReplicaThread and pull data from azure.
While BackupCheckManager would create BackupCheckerThread. And verification can go while we are recovering backups.